src/routers/project/projects.py

```diff
@@ -1,70 +1,139 @@
+    owned = request.headers["owned"]
+    participating = request.headers["participating"]
+    accessible = []
+    for li in [owned, participating]:
+        if len(li.rstrip()) != 0:
+            accessible = accessible + li.split(" ")
+    try:
+        return project_handler.get_all(db, accessible)
     except HTTPException as ex:
         raise ex
 
 
-@project_router.post("/projects", response_model=Project)
-async def make_new_project(new_project: NewProject,
+@project_router.post("/projects")
+async def make_new_project(request: Request,
+                           new_project: NewProject,
                            db: Session = Depends(get_db),
                            project_handler: object = Depends(createHandler)):
+    user_calling = request.headers["username"]
     try:
         return project_handler.create(
-            new_project.name,
-            new_project.created_by,
-            new_project.description,
-            db
+            name=new_project.name,
+            created_by=user_calling,
+            description=new_project.description,
+            db=db
         )
     except HTTPException as ex:
         raise ex
 
 @project_router.get("/project/{project_id}/info", response_model=Project)
-async def get_project_details(project_id: int,
+async def get_project_details(request: Request,
+                              project_id: int,
                               db: Session = Depends(get_db),
                               project_handler: object = Depends(createHandler)):
+    # check if project exists
+    try:
+        project_handler.check_project_exists(project_id, db)
+    except HTTPException as ex:
+        raise ex
+    owned = request.headers["owned"]
+    participating = request.headers["participating"]
+    # check appropriate privileges
+    if str(project_id) not in owned.split(" ") and str(project_id) not in participating.split(" "):
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN,
+                            detail="You don't have access to this project.")
     try:
         return project_handler.get(project_id, db)
     except HTTPException as ex:
         raise ex
 
 
 @project_router.put("/project/{project_id}/info", response_model=Project)
-async def update_project_details(project_id: int,
+async def update_project_details(request: Request,
+                                 project_id: int,
                                  new_info: UpdateProject,
                                  db: Session = Depends(get_db),
                                  project_handler: object = Depends(createHandler)):
-    if new_info.model_fields_set == set("updated_by"):
+    if new_info.model_fields_set == set():
         raise HTTPException(
             status_code=400,
             detail="No project properties were specified in the request body"
         )
+    # check if project exists
+    try:
+        project_handler.check_project_exists(project_id, db)

```

Comment:
It is a bad practice to catch all  exceptions in general (but the most fast way I know)
Since you wrap this call with try-except everywhere, you can raise httpexception inside of check_project_exists if the project does not exist (if you don't do it yet, gitlab does not want to show the code of this method)
If you worry that smth can go wrong during the db querying - raise httpexception on the level where this querying happens